### PR TITLE
Start using bom

### PR DESCRIPTION
**Pull request by joseblas**
start using bom dependencies

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -67,67 +67,67 @@
...
-            <version>2.16</version>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>scm-api</artifactId>
-            <version>2.2.6</version>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-support</artifactId>
```
Comment (jglick) [inline pom.xml]: Probably was here only to manage a version.

```suggestion
```

File: pom.xml [hunk 2]
```diff
@@ -67,67 +67,67 @@
...
         <dependency>  <!-- For Semaphore step -->
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-support</artifactId>
-            <version>${workflow-support-plugin.version}</version>
             <classifier>tests</classifier>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-scm-step</artifactId>
```
Comment (jglick) [inline pom.xml]: probably this too
```suggestion
```

File: pom.xml [hunk 3]
```diff
@@ -67,67 +67,67 @@
...
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-durable-task-step</artifactId>
-            <version>2.22</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>script-security</artifactId>
```
Comment (jglick) [inline pom.xml]:
```suggestion
```

**Comment by joseblas** [comment]
@jglick should I try to flatten the dependencies? remove all the unnecessary ones?

**Comment by jglick** [comment]
> remove all the unnecessary ones?

This. Sometimes a `dependency` was in a POM merely to satisfy `RequireUpperBoundDeps`; it was not being used as such. In these cases it should always have been in `dependencyManagement`, and can now be removed if the BOM declares it.

**Comment by joseblas** [comment]
sorry @jglick, I got confused.

File: pom.xml [hunk 4]
```diff
@@ -67,80 +67,56 @@
...
         </dependency>
-        <dependency>
-            <groupId>org.jenkins-ci.plugins.workflow</groupId>
-            <artifactId>workflow-scm-step</artifactId>
-            <version>2.4</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.jenkins-ci.plugins.workflow</groupId>
-            <artifactId>workflow-durable-task-step</artifactId>
```
Comment (jglick) [inline pom.xml]: This is definitely being used in tests and so ought to remain as an explicit dependency, even if it happens to also be pulled in from something else.

Comment (joseblas) [reply pom.xml]: Is a test dependency, I guess as far as tests are passing, is all good.

Comment (jglick) [reply pom.xml]: Looks like it is being pulled in via `workflow-basic-steps`. Just poor style generally to rely on transitive dependencies for things actually used directly. Anyway yes, if a future update of `workflow-basic-steps` drops that dep (as is quite possible in this case), updating to that would (I hope!) cause tests to start failing until you readded the explicit `workflow-durable-task-step` dep.

Comment (joseblas) [reply pom.xml]: I'd say `workflow-basic-steps` is directly used and its dependencies. But, let me know if you want this added back.

Comment (jglick) [reply pom.xml]: The `node` step is used in a few tests. Not important, just FYI.

**Pull request merged**